contracts/src/core/AlignedLayerServiceManager.sol

```diff
@@ -141,26 +153,25 @@ contract AlignedLayerServiceManager is
 
         emit BatchVerified(batchMerkleRoot, senderAddress);
 
-        // Calculate estimation of gas used, check that batcher has sufficient funds
-        // and send transaction cost to aggregator.
-        uint256 finalGasLeft = gasleft();
-
         // 70k was measured by trial and error until the aggregator got paid a bit over what it needed
-        uint256 txCost = (initialGasLeft - finalGasLeft + 70000) * tx.gasprice;
+        uint256 txCost = (initialGasLeft - gasleft() + 70_000) * tx.gasprice;
 
-        if (batchersBalances[senderAddress] < txCost) {
+        if (txCost > currentBatch.maxFeeAllowedToRespond) {
             revert InsufficientFunds(
                 senderAddress,
-                txCost,
-                batchersBalances[senderAddress]
+                currentBatch.maxFeeAllowedToRespond,
+                txCost
```

Comment:
this should raise another error, like "ExceededMaxFee" or something similar

Reply:
solved